### PR TITLE
JsValue inputs now are forced to be Strings

### PR DESCRIPTION
**Pull request by andrewconner**

### CODE REVIEW

File: shoebox/app/com/keepit/common/cache/FortyTwoCache.scala [hunk 1]
```diff
@@ -137,9 +138,23 @@ trait FortyTwoCache[K <: Key[T], T] extends ObjectCache[K, T] {
...
+      case x: java.lang.Short => x.shortValue()
+      case x: java.lang.Integer => x.intValue()
+      case x: java.lang.Long => x.longValue()
+      case x: java.lang.Float => x.floatValue()
+      case x: java.lang.Double => x.doubleValue()
+      case x: java.lang.Character => x.charValue()
+      case x: java.lang.Boolean => x.booleanValue()
+      case x: scala.Array[_] => x
+      case x: JsValue => Json.stringify(x)
+      case x => x
```
Comment (2is10) [inline shoebox/app/com/keepit/common/cache/FortyTwoCache.scala:153]: Shouldn't it fail if the type is not a primitive, a string, a byte array, or a `JsValue`?

Comment (andrewconner) [inline shoebox/app/com/keepit/common/cache/FortyTwoCache.scala:153]: Hm. Yeah that might be a good thing to do. I will need to verify that we aren't persisting objects everywhere else before I'm confident that it won't break things. Added to the list.

**Comment by andrewconner** [comment]
This will be included in a larger commit.

File: shoebox/app/com/keepit/common/cache/FortyTwoCache.scala (hunk 1 above)
Comment (eishay) [inline shoebox/app/com/keepit/common/cache/FortyTwoCache.scala:153]: So maybe send a healthckeck error if it happens. We don't have too many places to check, right?

**Pull request closed without merging**